### ISSUE TITLE
Keyboard shortcuts for editor (HW)

### ISSUE THREAD

**Issue by gsantner** [gsantner/markor#154]
#### General information

* **App version:** 0.2.5
* **System:** 

#### Description
Add keybindings like Ctrl-Z, Ctrl-Y, Ctrl-A, Ctrl-S

https://stackoverflow.com/questions/12250563/custom-keyboard-shortcuts

**Comment by gsantner** [comment]
The basics for this implemented, just calls to the methods need to happen ;)

**Comment by conrad-heimbold** [comment]
CTRL+V (Paste), 
CTRL+C (Copy), 
CTRL+X (Cut), 
CTRL+Z (undo), 
CTRL+A (mark all) 
... all seem to work. 

Only CTRL+S (save) is missing.

**Comment by gsantner** [comment]
CTRL+S can be mapped to the save method. Estimated this is like 10 lines. Contributions welcome!

**Comment by gsantner** [comment]
Long pressing the 'Special key' button will jump to bottom/top of document in next release.

**Comment by sojusnik** [comment]
Shortcuts for Markdown formatting would be very useful too:

H1, H2, H3: Ctrl + 1-3
Italic: Ctrl + i
Bold: Ctrl + b
Quote: Ctrl + q
List & Numbered List: Ctrl + l

If the text cursor is at the beginning, end or in the middle of a word, then pressing the above mentioned shortcut should format the word correspondingly.

**Comment by mariellep** [comment]
Ability to set some of the existing shortcuts to a Bluetooth keyboard mapping would be useful. For example, indent/de-indent requires hitting the keyboard button in the editor and scrolling down as opposed to just hitting tab on a regular computer editor. Perhaps ctrl+tab for indent and ctrl+shift+tab for de-indent?

**Comment by gsantner** [comment]
Yes thats among what I thought too. But should work on any keyboard, shouldnt matter that its BT

**Comment by aisbergde** [comment]
I have no idea how to use some "Ctrl + ..." combination on android. I have no Ctrl on any of my keyboards.
What I mean is one more entry in this list, to be able to mark a line:

![Screenshot_2020-05-24-01-22-07-470_net gsantner markor](https://user-images.githubusercontent.com/9020187/82742357-531ab500-9d5d-11ea-86f7-20d22c29d19c.jpg)

**Comment by gsantner** [comment]
Thanks now I know where you mean. Is quite easy to add, so when somebody reads this: for functionality it's two invocations, plus if+string.

**Comment by gsantner** [comment]
I've mixed up the thread, moved comment to original thread.

**Comment by biggles5107** [comment]
I'm requesting that <kbd>Tab</kbd> and <kbd>Shift</kbd><kbd>Tab</kbd> be mapped to indent and de-indent, respectively.

**Comment by harshad1** [comment]
We have tab support now. Closing this

**Comment by orionlee** [comment]
Arguably, there are still some issues (possibly tracked as separate issues?). As of v2.14.0 :
1. there is no keyboard shortcut for Search (the new tab support is not applicable for Search button, at the top action bar)
2. for the new tab support,  
  a. If users enable the setting "Edit mode > Indent lines with TAB key": tab / shift-tab will indent / de-indent the line.
  b. If the setting is disabled: tab will jump to the bottom action bar.
  Is that the intended behavior?
  As one data point, I choose (a) (indent / de-indent). Since I want to preserve (a), I still can't get to the bottom action bar using a keyboard.

**Comment by harshad1** [comment]
That's the expected behavior. Tab is what android uses to jump to the next view. We now have the option to intercept it and indent instead

**Comment by alioth9** [comment]
Hi first thanks you for the work on this so useful app maintaining and improving it.
There is one keyboard shortcut which could be useful (I didn't see it implemented ) : for switching between Edit and Reading mode, I suggest `Insert` (interesting for me) and maybe something else with Ctrl (maybe more expected). 

Since it is not exactly a shortcut for edit context I could create anolther issue if you prefer.

**Comment by gsantner** [comment]
Additions are welcome.

**Comment by alioth9** [comment]
My bad I didn't see there was one added : `C-/`

**Comment by gsantner** [comment]
Update: see https://github.com/gsantner/markor/pull/2733
